pkg/network/master/egressip.go

```diff
@@ -121,11 +108,11 @@ func (eim *egressIPManager) maybeDoUpdateEgressCIDRs() (bool, error) {
 		}
 	}
 
-	eim.monitorNodes = monitorNodes
+	eim.tracker.MonitorNodes = monitorNodes
```

Comment:
I agree here, but it's not that evident. Master and node have two different set of rules for how they wish to monitor nodes. Nodes only decide to monitor when the egress IP is not assigned to the specific node in question, and master only monitors when `egressCIDRs` are set (and subsequently stops monitoring when the assigned egress IPs, calculated from `egressCIDRs`, is empty). I am not sure we can have `common` manage that by itself, and even if it would be possible, it might require more refactor than we're willing to do. What I can do to avoid master and node tampering with common's data, is: define some public methods (`SetMonitorNodes` for master, `AddEgressIP`/`RemoveEgressIP` for node) which would do what master and node already do, but inside common instead.